src/App.jsx

```diff
@@ -14,45 +18,130 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHA_ASC = 'asc';
+const SORT_BY_ALPHA_DESC = 'desc';
+const SORT_BY_LENGTH_ASC = 'up';
+const SORT_BY_LENGTH_DESC = 'down';
+const REVERSE = 'reverse';
+const RESET = 'reset';
+
+function sortingFunction(data, sortField, prevValue) {
+  const list = [...data];
+  const reverseData = [...prevValue];
+
+  if (sortField === RESET) {
+    return list;
+  }
+
+  if (sortField !== REVERSE) {
+    list.sort((a, b) => {
+      switch (sortField) {
+        case SORT_BY_ALPHA_ASC:
+          return b.localeCompare(a);
+        case SORT_BY_ALPHA_DESC:
+          return a.localeCompare(b);
+        case SORT_BY_LENGTH_ASC:
+          return a.length - b.length;
+        case SORT_BY_LENGTH_DESC:
+          return b.length - a.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return sortField === REVERSE ? reverseData.reverse() : list;
+}
+
+function displayingResset(initial, current) {
+  return JSON.stringify(initial) === JSON.stringify(current);
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [isAlpha, setAlptha] = useState(false);
+  const [isLength, setLength] = useState(false);
+  const [isReverse, setReverse] = useState(false);
+
+  const [isSorted, buttonAction] = useState('');
```

Comment:
The names 'isSorted' and 'buttonAction' are not clear. Consider renaming them to reflect their purpose.